File: elemental/src/dto/postdto.ts
import {UserDTO} from "./userdto";

export class PostDTO {
  idPost: number

  dataPost: Date

  contenuto: string

  imgURL: string

  user: UserDTO

  constructor(idPost: number,

  dataPost: Date,

  contenuto: string,


  imgURL: string,

  user: UserDTO) {
    this.idPost = idPost
    this.dataPost = dataPost
    this.contenuto = contenuto
    this.imgURL = imgURL
    this.user = user
  }
}

File: src/main/java/com/simonecampis/Elemental/dto/PostDTO.java
package com.simonecampis.Elemental.dto;

import com.simonecampis.Elemental.model.User;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.sql.Date;
@Data
@NoArgsConstructor
@AllArgsConstructor
public class PostDTO {
    private Long idPost;

    private Date dataPost;

    private String contenuto;

    private String imgURL;

    private User user;
}
